.github/workflows/rust.yaml

```diff
@@ -83,28 +83,44 @@ jobs:
         run: ./scripts/ci_build_rust.bash
         shell: bash
 
-      - uses: actions/upload-artifact@v2
-        with:
-          name: ${{ steps.build.outputs.artifact_name }}
-          path: ${{ steps.build.outputs.artifact_name }}
-
-  publish:
-    if: startsWith(github.ref, 'refs/tags/v')
-    needs: [build]
-    name: Publish github release
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v2
-
-      - uses: actions/download-artifact@v2
-        with:
-          path: artifacts
+      - name: Cache artifact names for package registry
+        if: startsWith(github.ref, 'refs/tags/v')
+        shell: bash
+        run: |
+          if [ ${{ matrix.os }} == "ubuntu-18.04" ]; then
+            echo "::set-output name=LINUX_RELEASE_ARCHIVE::${{ steps.build.output.artifact_name }}"
+          elif [ ${{ matrix.os }} == "macos-12" ]; then
+            echo "::set-output name=MAC_RELEASE_ARCHIVE::${{ steps.build.output.artifact_name }}"
+          elif [ ${{ matrix.os }} == "windows-2022" ]; then
+            echo "::set-output name=WINDOWS_RELEASE_ARCHIVE::${{ steps.build.output.artifact_name }}"
+          fi
 
       - name: Upload Release
+        if: startsWith(github.ref, 'refs/tags/v')
         uses: svenstaro/upload-release-action@v1-release
         with:
           repo_token: ${{ secrets.GITHUB_TOKEN }}
           file: "artifacts/*/*.zip"
           tag: ${{ github.ref }}
           overwrite: true
           file_glob: true
+
+  registry:
+    if: startsWith(github.ref, 'refs/tags/v')
+    needs: [ build ]
+    name: Publish rust release to package-registry
+    runs-on: ubuntu-latest
+    steps:
+      - name: Hook into package registry
+        uses: swift-nav/swift-cli-publisher@v2
+        env:
+          DL_LINUX: ${{ needs.build.outputs.LINUX_RELEASE_ARCHIVE }}
```

Comment:
Seems like we can't access the output of previous jobs if it is dynamically modified (as per https://stackoverflow.com/questions/59175332/using-output-from-a-previous-job-in-a-new-one-in-a-github-action as well)

Reply:
Per discussion we probably can share data by using well known artifact names then uploading/downloading in different jobs, please re-request a review once you have that working

Reply:
Just switched them both to enforce binary targets and hardcoded in the paths